web/clusters.go

```diff
@@ -81,18 +87,13 @@ func NewClusterListHandler(clustersService services.ClustersService) gin.Handler
 			return
 		}
 
-		healthContainer := NewClustersHealthContainer(clusterList)
+		healthContainer := NewClustersHealthContainer(clusterListAll)
 		healthContainer.Layout = "horizontal"
 
-		count, err := clustersService.GetCount()
-		if err != nil {
-			_ = c.Error(err)
-			return
-		}
-		pagination := NewPagination(count, pageNumber, pageSize)
+		pagination := NewPagination(len(clusterListAll), pageNumber, pageSize)
```

Comment:
I would instead introduce another health count service method that gives as an output the health counts directly, and the delegate that to the service instead of loading again the whole list in the controller, and use the count method for the pagination lenght.

Reply:
But what would this improve? In order to get the health numbers, we need to get all the clusters with the applied filters and get the health data. 
We could create other service method, but it would do exactly the same as far as I see. I don't know any way to save any query to the database. 
It might look cleaner, yes. But the code efficiency and in fact almost all the code would be the same.
Or am I missing something?

PD: And we cannot use the currently existing `GetCount` method, as we need to get the count with the filtered values, which at the end, would require to use the same sql where conditions we have now. Does it make sense to complicate the code yet more?

Reply:
PD2: The unique way I see to reduce this to a unique call is to do the pagination within the code instead of using sql conditions

Reply:
> It might look cleaner, yes. But the code efficiency and in fact almost all the code would be the same.
Or am I missing something?

If I am not mistaken, I think that to get the health status we don't need a "select *" query that also does various preload (e.g. Tags) but just the IDs of  the clusters (a select id + pluck) should be enough to be used in the check service `GetAggregatedChecksResultByCluster(cluster.ID)`

> PD: And we cannot use the currently existing GetCount method, as we need to get the count with the filtered values, which at the end, would require to use the same sql where conditions we have now.

By abstracting the filtering to a new method we can shape a GetCount with the filters, I don't think this would complicate the code a lot.

Reply:
No, the get the health we need to preload the tags, as we need to filter them. And we would need to get the `id`, `name`, `cluster_type` and `sid`. And we would need to get the checks results too. So the unique difference would to we discard only the number of hosts and the number of resources